R/build-redirects.R

```diff
@@ -0,0 +1,47 @@
+build_redirects <- function(pkg = ".",
+                            override = list()) {
+  pkg <- section_init(pkg, depth = 1L, override = override)
+
+  if (is.null(pkg$meta$redirects)) {
+    return(invisible())
+  }
+
+  rule("Building redirects")
+  if (is.null(pkg$meta$url)) {
+    abort(sprintf("Can't find %s.", pkgdown_field(pkg, "url")))
+  }
+
+
+  purrr::iwalk(
+    pkg$meta$redirects,
+    build_redirect,
+    pkg = pkg
+  )
+}
+
+build_redirect <- function(entry, index, pkg) {
+  if (length(entry) != 2) {
+    abort(
+      sprintf(
+        "All redirect entries must have length 2 (entry %s in %s has length %s).",
```

Comment:
Maybe something more like this:

```suggestion
        "%s[[%s]] must be a character vector of length 2.", pkgdown_field(pkg, "redirects"), index
```